### ISSUE TITLE
Unblock ads

### ISSUE THREAD

**Issue by tcanbot** [TeamNewPipe/NewPipe#1819]
Content creators on YouTube were severely affected by the adpocalypse. I really like this application and I want to use it but I don't want to hurt anyone's ad revenue. I don't know if it's even possible because it doesn't seem like it is. This application works very well on older/less efficient hardware so it was great for an Android TV without play services because it seems nothing can be done on the new version of YouTube without play services. I have a shitty pre-installed version but it's only good for glitchy low quality playback and the speed cannot be changed. On newpipe I can change the pitch with or without changing the playback speed so it's a great application but I would love it if we could use it while continuing to support the ad dependent people making videos on YouTube.

**Comment by gkeegan** [comment]
I agree slightly. Perhaps an opt-in option would be better? Setting to show ads only on specific channels or something like that? It would also be difficult to serve YouTube ads I think, as it would have to load the ad, send watch time of the ad to YouTube and some other things most likely.

**Comment by kowith337** [comment]
If it so, the developers will have to create a different branch because F-Droid will refuse to publish the app due to it have third party ads (an also get flagges that have `Ads` antiuser as well) like SkyTube have F-Droid version and Extra version that can change internal video player to use official YouTube API Player. However, if you have official app installed and not disabled (bloated pre-install or install GApps in custom rom later, etc.)

**Comment by theScrabi** [comment]
Right. The problem is, if we used the youtube api we could not make NewPipe as awesome as it is, because having an api acces toke legally requires you to restrict functionality. Thus advertisement never works.

However I am aware of the problem that creates don't earn money by NewPipe, this is why once I wanted to add a special button that highlites if affiliate or donation links are put in the videodescription, so you can grand them money if you buy something on amazon.
The code almost made it in, but i threw it out in the end because there where several issues with link handling in descriptions back then.

However if there is still interest for highlited donation button, I'd be happy to add this :)

**Comment by tcanbot** [comment]
@gkeegan yeah I think sending the watch time of the ad would be especially crucial otherwise the ad would be as good as blocked because some downloaders can inadvertently function as ad blockers if they start downloading an ad instead of letting it play in the browser which is why this doesn't seem doable. Your comment about opting in for certain channels reminded me of the brave browser. It has a point based system for allotting points to various channels and the allotment of these points will then somehow influence they're ad revenue.

**Comment by tcanbot** [comment]
@kowith337 I had downloaded it from f-droid. It was the first result that came up when I had searched for YouTube on f-droid but I didn't consider that it would be flagged differently because of the ads and the tracking. Google ads track everywhere they're present. I remember once I had used a different search engine to visit a site but I could still find that page on my activity on Google because a Google banner ad was being displayed on that page so Google could track my activity even though I didn't visit any of the Google sites. I will have to lookup skytube and that example does explain this situation very well.

**Comment by gkeegan** [comment]
@theScrabi I think your idea of having donation buttons would be much better than the ads actually. This way the money would go directly to them, and YouTube cannot take it away through demonetization.

**Comment by kowith337** [comment]
@tcanbot I mean about flagging NonFree `Ads` and `Tracking` is it possible if deploy a version that have ads, but F-Droid will be strictly refuse and none of anything can be public on F-Droid at all.
For finding extra version of SkyTube, you need to find it outside F-Droid or go to release page of their source code that you can download an extra version directly.

**Comment by theScrabi** [comment]
We will not do that.

**Comment by trymeouteh** [comment]
I against this for a few reasons.

1. NewPipe is the best YT app out there (I Don't care what anyones says)

2. NewPipe is great because it respects users privacy because block ads and trackers

3. I have a love/hate relationshit with YT. I love the videos on YouTube but I do not like their censorship so by blocking ads is my way of revolting against YT which causes YT not to make revenue. The only con about this is that the content creator also doesn't make any ad revenue. But if you like a YT Content Creator you should support them via Patreon, Bitcoin, etc.

Also many YouTubers are using sponsors in their videos which act as ads within the video. These "ads" can be skipped by skipping 10, 20 seconds into the video.

**Comment by KonoromiHimaries** [comment]
it can be work like https://github.com/dhowe/AdNauseam to safe privacy, but ethic.

**Comment by Stypox** [comment]
@KonoromiHimaries would that mean "clicking on" (i.e. making a network request to) every ad provided by Youtube in the background?

**Comment by KonoromiHimaries** [comment]
https://github.com/TeamNewPipe/NewPipe/issues/2554

**Comment by TheAssassin** [comment]
The title of this issue is technically wrong. NewPipe doesn't block ads at all. If YouTube were embedding ads in their videos (which they hopefully won't do) and we'd cut them out, that'd qualify for logging. However, we just fetch the videos as-is from YouTube.

What you suggest is implementing something like their ads stuff, which is basically, playing a separate video before and in the middle of another video.

It is not our job to implement Google's ads spyware crap in our free and open source app. It'd be a usability nightmare, too.

I think we should close this issue.

**Comment by Heath123** [comment]
I thought I'd found a good open-source YouTube app, but I can't use it without feeling bad because of this... Can't it at least be considered, as an option?

**Comment by TheAssassin** [comment]
Feel free to fork the app and implement YouTube tracking as a therapy. But note you have to change the name from NewPipe to something else.
